remoteappmanager/handlers/base_handler.py

```diff
@@ -40,3 +41,20 @@ def render(self, template_name, **kwargs):
 
         args.update(kwargs)
         super(BaseHandler, self).render(template_name, **args)
+
+    def write_error(self, status_code, **kwargs):
+        """Render error page for uncaught errors"""
+        status_message = responses.get(status_code, 'Unknown HTTP Error')
+        message = ""
+
+        # If this error was caused by an uncaught exception
+        # log exception message and reference number as well
+        exc_info = kwargs.get('exc_info')
+        if exc_info:
+            exception = exc_info[1]
+            ref = self.log.issue(status_message, exception)
+            reason = getattr(exception, 'reason', '')
+            message = '{} Ref.: {}'.format(reason, ref)
```

Comment:
which brings us back to the `reason` discussion. Is it "proper" to change the HTTP reason, when it's supposed to be a standard string? What are others doing? Tornado documentation says that reason is specified only when using non standard codes, not that we can put anything in it.

I am not saying it's not possible, just if it's good practice, and from what I know of HTTP, it isn't